src/Event/AfterRun.php

```diff
@@ -0,0 +1,22 @@
+<?php
+declare(strict_types=1);
+
+namespace Yiisoft\Widget\Event;
+
+/**
+ * AfterRun event is raised right after executing a widget.
+ */
+class AfterRun
+{
+    private string $result;
+
+    public function __construct(string $result)
```

Comment:
How about making the event Widget-aware? It might be useful to access the Widget context when dealing with the result of its execution

```suggestion
    public function __construct(Widget $widget, string $result)
```

Reply:
That is an interesting idea. Any exaples?

Reply:
The problem is a bit wider, actually. In Yii2 we could do something like this:

```php
\thirdparty\acme\Avatar::widget([
    'email' => $email,
    'on afterRun' => function (WidgetEvent $event): void {
        if ($event->result === '') {
             $event->result = \ourcompany\AvatarPlaceholder::widget();
        }
    }
]);
```

(Yes, I know, it's a dumb example because we can write it much shorter)

```php
\thirdparty\acme\Avatar::widget(['email' => $email]) ?: \ourcompany\AvatarPlaceholder::widget();
```

Nevertheless, with a separate event handlers definition, we:

1. Are not able to consolidate knowledge about the single widget call closely in the code
2. Are not able to distinguish the event originator, making the event pretty much useless.

Maybe I did not get how to use the event handlers with this draft?

Reply:
Thought about it more. Added.